File: Practice/Data Structures/Linked List/Linked List-C++/Driver.cpp
/*
*	Driver.cpp by Randyll Bearer 2018
* 	This file was written in order to test its corresponding Node.h implementation
*	in order to test a doubly-linked list data structure.
*/

//Library Files
#include <iostream>

//Header Files
#include "Node.h"

int main(){
	
	std::cout << "Initializing nodes and linked list\n";
	
	//Initialize Three Nodes
	Node first = Node(1);
	Node second = Node(2);
	Node third = Node(3);
	
	//Combine the Nodes into a linked list
	first.next = &second;
	second.next = &third;
	second.previous = &first;
	third.previous = &second;
	
	//Print all nodes
	std::cout << "Printing from first node...\n";
	first.printFromHere(&first);
	
}
//End of File

File: Practice/Language-Specific/Python/Linked List-PY/Driver.py
# Driver.py by Randyll Bearer 2018
# Basic program to test a doubly-linked list structure using the
# corresponding Node.py implementation.

import Node

class Driver:

	#Main Driver
	def main():
		#Initialize three nodes
		first = Node.Node()
		first.value = 1
		second = Node.Node()
		second.value = 2
		third = Node.Node()
		third.value = 3
		
		#Create doubly-linked list
		first.next = second
		second.next = third
		second.previous = first
		third.previous = second
		
		#Print elements from first
		first.printFromHere()
	
	
	if __name__ == '__main__':
		main()

#End of File